bdb/bdb_blkseq.c

```diff
@@ -726,7 +726,13 @@ int bdb_recover_blkseq(bdb_state_type *bdb_state)
                     if (rectype == DB_llog_ltran_commit) {
                         bp = (logdta.data + sizeof(u_int32_t) + sizeof(u_int32_t));
                         LOGCOPY_TOLSN(&bslsn, bp);
-                        rc = logc->get(logc, &bslsn, &logdta, DB_SET);
+
+                        /* Purge-old-files emits empty transactions in snapshot */
+                        if (IS_ZERO_LSN(bslsn)) {
+                            rc = -1;
```

Comment:
Do we need to continue here?

Reply:
This is okay, actually. rc is reset to 0 by the logc->get(DB_SET) call at the end of the loop, and we'll be able to continue the loop.